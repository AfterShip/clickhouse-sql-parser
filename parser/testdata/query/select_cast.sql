select cast(1 as Float64) as value;
select cast(1, 'Float64') as value;
select (1 as Float64) as value;
select 1::Float64 as value;