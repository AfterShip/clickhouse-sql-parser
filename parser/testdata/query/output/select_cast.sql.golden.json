[
  {
    "SelectPos": 0,
    "StatementEnd": 34,
    "With": null,
    "Top": null,
    "HasDistinct": false,
    "SelectItems": [
      {
        "Expr": {
          "CastPos": 7,
          "Expr": {
            "NumPos": 12,
            "NumEnd": 13,
            "Literal": "1",
            "Base": 10
          },
          "Separator": "as",
          "AsPos": 14,
          "AsType": {
            "Name": {
              "Name": "Float64",
              "QuoteType": 1,
              "NamePos": 17,
              "NameEnd": 24
            }
          }
        },
        "Modifiers": [],
        "Alias": {
          "Name": "value",
          "QuoteType": 1,
          "NamePos": 29,
          "NameEnd": 34
        }
      }
    ],
    "From": null,
    "ArrayJoin": null,
    "Window": null,
    "Prewhere": null,
    "Where": null,
    "GroupBy": null,
    "WithTotal": false,
    "Having": null,
    "OrderBy": null,
    "LimitBy": null,
    "Limit": null,
    "Settings": null,
    "Format": null,
    "UnionAll": null,
    "UnionDistinct": null,
    "Except": null
  },
  {
    "SelectPos": 36,
    "StatementEnd": 70,
    "With": null,
    "Top": null,
    "HasDistinct": false,
    "SelectItems": [
      {
        "Expr": {
          "CastPos": 43,
          "Expr": {
            "NumPos": 48,
            "NumEnd": 49,
            "Literal": "1",
            "Base": 10
          },
          "Separator": ",",
          "AsPos": 49,
          "AsType": {
            "LiteralPos": 52,
            "LiteralEnd": 59,
            "Literal": "Float64"
          }
        },
        "Modifiers": [],
        "Alias": {
          "Name": "value",
          "QuoteType": 1,
          "NamePos": 65,
          "NameEnd": 70
        }
      }
    ],
    "From": null,
    "ArrayJoin": null,
    "Window": null,
    "Prewhere": null,
    "Where": null,
    "GroupBy": null,
    "WithTotal": false,
    "Having": null,
    "OrderBy": null,
    "LimitBy": null,
    "Limit": null,
    "Settings": null,
    "Format": null,
    "UnionAll": null,
    "UnionDistinct": null,
    "Except": null
  },
  {
    "SelectPos": 72,
    "StatementEnd": 102,
    "With": null,
    "Top": null,
    "HasDistinct": false,
    "SelectItems": [
      {
        "Expr": {
          "LeftParenPos": 79,
          "RightParenPos": 92,
          "Items": {
            "ListPos": 80,
            "ListEnd": 92,
            "HasDistinct": false,
            "Items": [
              {
                "Expr": {
                  "NumPos": 80,
                  "NumEnd": 81,
                  "Literal": "1",
                  "Base": 10
                },
                "Alias": {
                  "Name": "Float64",
                  "QuoteType": 1,
                  "NamePos": 85,
                  "NameEnd": 92
                }
              }
            ]
          },
          "ColumnArgList": null
        },
        "Modifiers": [],
        "Alias": {
          "Name": "value",
          "QuoteType": 1,
          "NamePos": 97,
          "NameEnd": 102
        }
      }
    ],
    "From": null,
    "ArrayJoin": null,
    "Window": null,
    "Prewhere": null,
    "Where": null,
    "GroupBy": null,
    "WithTotal": false,
    "Having": null,
    "OrderBy": null,
    "LimitBy": null,
    "Limit": null,
    "Settings": null,
    "Format": null,
    "UnionAll": null,
    "UnionDistinct": null,
    "Except": null
  },
  {
    "SelectPos": 104,
    "StatementEnd": 130,
    "With": null,
    "Top": null,
    "HasDistinct": false,
    "SelectItems": [
      {
        "Expr": {
          "LeftExpr": {
            "NumPos": 111,
            "NumEnd": 112,
            "Literal": "1",
            "Base": 10
          },
          "Operation": "::",
          "RightExpr": {
            "Name": "Float64",
            "QuoteType": 1,
            "NamePos": 114,
            "NameEnd": 121
          },
          "HasGlobal": false,
          "HasNot": false
        },
        "Modifiers": [],
        "Alias": {
          "Name": "value",
          "QuoteType": 1,
          "NamePos": 125,
          "NameEnd": 130
        }
      }
    ],
    "From": null,
    "ArrayJoin": null,
    "Window": null,
    "Prewhere": null,
    "Where": null,
    "GroupBy": null,
    "WithTotal": false,
    "Having": null,
    "OrderBy": null,
    "LimitBy": null,
    "Limit": null,
    "Settings": null,
    "Format": null,
    "UnionAll": null,
    "UnionDistinct": null,
    "Except": null
  }
]